cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -124,6 +128,66 @@ func NewFromStaticPod(endpoints []string, manifestDir string, certificatesDir st
 	return New(endpoints, "", "", "")
 }
 
+// NewFromPods creates a GenericClient for the pods in the namespace kube-system and with label
+// component=etcd, manifestDir, and certificatesDir
+func NewFromPods(client clientset.Interface, manifestDir string, certificatesDir string) (*Client, error) {
+
+	// Gets the list of pods
+	glog.V(1).Infoln("getting the list of pods with component=etcd")
+	listOpts := metav1.ListOptions{LabelSelector: "component=etcd"}
+	pods, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(listOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	// Note for reviewers: I'm not sure this is the best method for getting the endpoint
```

Comment:
I think this is a good approach with one minor concern: there is a race condition that the PodIP may change after the PodIP lookup and before our client access, but I'm ok living with that possibility.

I wonder if there is a clean way to solve this by managing etcd as a statefulset. The other thought I had was adding a Service object to the static manifest and managing that by hand (by kubeadm). The goal of these two ideas is to provide a stable DNS name instead of a PodIP lookup. Either way, those types of changes would be way out of scope for this PR.

Reply:
So the ClusterStatus should have the endpoints. 
Or as discussed on the call put the meta-data in an annotation for the pod to make it easier to extract.

Reply:
+1 to using ClusterStatus or a pre-set annotation to use as a starting point.

After we have the initial endpoint list, we should use that to query the etcd cluster itself for the list of endpoints.

Reply:
Potentially, we should also raise an error if the queried endpoints do not match the expected endpoints, or if the cluster is not in a healthy state to start.

Reply:
done reading from cluster status + calling sync to get the real list of endpoints from etcd